custom_components/sector/event.py

```diff
@@ -0,0 +1,140 @@
+"""Event platform for Sector Alarm integration."""
+
+import logging
+from datetime import datetime, timezone
+from collections import defaultdict
+from homeassistant.components.event import EventEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .coordinator import SectorAlarmConfigEntry, SectorDataUpdateCoordinator
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SectorAlarmConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+):
+    """Set up event entities based on processed data in Sector Alarm coordinator."""
+    coordinator: SectorDataUpdateCoordinator = entry.runtime_data
+    grouped_events = coordinator.process_events()
+
+    entities = []
+
+    for device_serial, events in grouped_events.items():
```

Comment:
I think we should skip this to always create the entities even if no logs are currently there

Reply:
It's going to be a lot of unused entities since the logs only cover locks/arming, this means we'll have loads of empty entities which will never get populated for cameras, smoke detectors, contact sensors etc.